pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -583,29 +457,50 @@ func (c *Compactor) runCompact(ctx context.Context, logger log.Logger, job Job,
 	}
 
 	if len(metas) == 0 {
-		// Get chunks data from list of chunkRefs
-		chks, err := storeClient.chunk.GetChunks(ctx, makeChunkRefs(job.Chunks(), job.Tenant(), job.Fingerprint()))
+		localDst := createLocalDirName(c.cfg.WorkingDirectory, job)
+		defer func() {
+			//clean up the bloom directory
+			if err := os.RemoveAll(localDst); err != nil {
+				level.Error(logger).Log(fmt.Sprintf("failed to remove block directory %s", localDst), err)
+			}
```

Comment:
Could a non-empty directory lead to a problem later on? Should we fail?

Reply:
No it shouldn't lead problems now, i don't think we should fail compaction for that. And how and where we do clean up will certainly change as we'll compress the blocks before uploading.